### PR TITLE
Validate ready status is removed for ovn when it's disabled

### PR DESCRIPTION
**Pull request by booxter**
This is built on top of the other PR from @gibizer that fixed the regression. This PR adds necessary test coverage (plus removes some redundant Remove calls).

### CODE REVIEW

**Comment by booxter** [comment]
The original fix (first patch in the series) is from https://github.com/openstack-k8s-operators/openstack-operator/pull/659

File: tests/functional/openstackoperator_controller_test.go [hunk 1]
```diff
@@ -197,6 +198,8 @@ var _ = Describe("OpenStackOperator controller", func() {
 					},
 				},
 			}
+			// TODO: had to disable tls to allow control plane status to update
+			spec["tls"] = map[string]interface{}{}
```
Comment (booxter) [inline tests/functional/openstackoperator_controller_test.go:202]: @stuggi any idea why I have to disable tls to allow the CP to get to Ready in functional tests? I see errors in the log about expires field being empty and not conforming to date-time format when the reconcile loop attempts to update Status. (and then the test case here fails because Ready states never get updated on CP).

**Comment by booxter** [comment]
recheck

File: tests/functional/openstackoperator_controller_test.go (hunk 1 above)
Comment (gibizer) [reply tests/functional/openstackoperator_controller_test.go:202]: I think we have a bug here https://github.com/openstack-k8s-operators/openstack-operator/blob/00c81bcd4ba46e1fcb12df2a62a22eb166ea7eb7/pkg/openstack/ca.go#L138-L143

The TLSCAStatus.Expires fields expects a data-time but the above code does not initialize it if there is no certs, so it ends up being initialized to the go empty value `""`. 

I put an extra log there and I see:
```
  2024-02-09T09:55:30.298+0100	INFO	!!! Setting endpoint status	{"controller": "openstackcontrolplane", "controllerGroup": "core.openstack.org", "controllerKind": "OpenStackControlPlane", "OpenStackControlPlane": {"name":"3d79d521-10c2-4707-8462-a","namespace":"ee00d7df-3130-4715-b928-58d93c968689"}, "namespace": "ee00d7df-3130-4715-b928-58d93c968689", "name": "3d79d521-10c2-4707-8462-a", "reconcileID": "d36a4742-3150-42f4-acfd-da70f7be3dcd", "endpoint": "public", "status": {"name":"rootca-public","expires":""}, "len(caOnlyBundle.certs)": 0}
```

Comment (gibizer) [reply tests/functional/openstackoperator_controller_test.go:202]: In the test the PEM data is "test" (in base64) and that result in an empty bundle due to breaking here
https://github.com/openstack-k8s-operators/openstack-operator/blob/00c81bcd4ba46e1fcb12df2a62a22eb166ea7eb7/pkg/openstack/ca.go#L397-L402

Comment (gibizer) [reply tests/functional/openstackoperator_controller_test.go:202]: so we either need a bit better test data with certs in it or we need to make the expires field to allow being empty (maybe a pointer type allow nil without forcing the date-time validation, I don't know)

Comment (gibizer) [reply tests/functional/openstackoperator_controller_test.go:202]: For this test case and this PR I'm OK to keep the TLS disabled.

File: tests/functional/openstackoperator_controller_test.go [hunk 2]
```diff
@@ -261,6 +280,12 @@ var _ = Describe("OpenStackOperator controller", func() {
 				instance := &ovnv1.OVNController{}
 				g.Expect(th.K8sClient.Get(th.Ctx, names.OVNControllerName, instance)).Should(Not(Succeed()))
 			}, timeout, interval).Should(Succeed())
+
+			// expect the ready status to become unknown since ovn is disabled
```
Comment (gibizer) [inline tests/functional/openstackoperator_controller_test.go:284]: The comment is wrong. We expect the OVN condition to be removed and the Ready condition to be True

File: tests/functional/openstackoperator_controller_test.go (hunk 1 above)
Comment (stuggi) [reply tests/functional/openstackoperator_controller_test.go:202]: we discussed in our tls meeting a change we want to make to the openstack-operator for the CA handling. I will rework on this in the next week or the following. I'll then also add more tests. for now we could disable it as in this pr

**Review by gibizer — APPROVED** [review]
thanks!

**Review by stuggi — APPROVED** [review]
/lgtm

**Comment by abays** [comment]
```
Removing debug pod ...
error: unable to upgrade connection: container container-00 not found in pod oko-14-lmt7f-master-1-debug-mnmpm_openstack-kuttl-tests
make[3]: *** [Makefile:526: crc_storage_cleanup] Error 1
```

/test openstack-operator-build-deploy-kuttl

**Comment by booxter** [comment]
recheck

**Review by fao89 — APPROVED** [review]
